Source/core/Singleton.cpp

```diff
@@ -36,8 +36,6 @@ namespace Core {
 
     /* virtual */ Singleton::SingletonList::~SingletonList()
     {
-        // Dispose was not called before main application was ended
```

Comment:
Had an internal discussion and the assumed problem did not occur there. Since the ASSERT is still reporting a real issue, I would like to continue to report on it. As you have a valid use case, do agree that ASSERTING in this usecase is not the preferred behavior as well. Could we than, as a compromise, turn this into a TRACE_L1(); Line so that if it is not disposed at the right time, people do get a warning log line.

Reply:
Maybe interesting to note, this assert was actually only indicating a real issue if Thunder is ran under windows. Windows has the nasty habit that, after reaching the end of the main, it first kills any created threads by the process before it starts calling the destructors of the static classes created during the runtime.
As in the destructor here, the Dispose is called anyway, meant that a lot of the destructors never completed, the closing down of the application hang, since, for example, the Thread object destructor waits till the thread reports "safe", but if the thread was killed (as it happens on windows) it never reached this "safe" state. The thread destructor, would than hang forever!
Hence why we opted for warning the developer to always chose a point, during normal execution (prior to leaving the main for example) to first call dispose.

Reply:
TRACE_L1 message introduced